PythonVisualizations/SimpleQueue.py

```diff
@@ -230,133 +264,127 @@ def display(self):
                 # increment xpos
                 xpos += CELL_SIZE
 
-        window.update()
+        self.rearArrow = self.createIndex(self.rear, "Rear", high=False)
+        self.frontArrow = self.createIndex(self.front, "Front", high=True)
+        self.window.update()
 
     #disable insert if queue if full, disable delete if empty
     #enable everything else without overriding queue/deque functionality
     def onOffButtons(self):
         #if it's a deque
-        if buttons[5]['relief'] == 'sunken':
-            enableButtons()
+        if self.buttons[5]['relief'] == 'sunken':
+            pass
+          #  enableButtons()
             
         #if it's a queue
         else:
-            buttons[0].config(state = NORMAL)
-            buttons[3].config(state = NORMAL)
+            self.buttons[0].config(state = NORMAL)
+            self.buttons[3].config(state = NORMAL)
         
         #in all cases: disable buttons as necessary
         if self.nItems == self.size:
-            buttons[0].config(state = DISABLED)
-            buttons[1].config(state = DISABLED)
+            self.buttons[0].config(state = DISABLED)
+            self.buttons[1].config(state = DISABLED)
             
         if self.nItems == 0:
-            buttons[2].config(state = DISABLED)
-            buttons[3].config(state = DISABLED)
+            self.buttons[2].config(state = DISABLED)
+            self.buttons[3].config(state = DISABLED)
```

Comment:
By only implementing the operations for a queue data structure, the enabling and disabling of buttons unique to queues and deques is not needed.